### PR TITLE
Canvas: Fix division by zero

### LINKED ISSUE #85031: Canvas: Connection bug when anchors line up perfectly
When attempting to modify a connection that is drawn between two anchors of exactly the same absolute x or y coordinates, an error occurs because division by zero leads to svg attributes of infinity: 

When creating a new vertex...
https://github.com/grafana/grafana/blob/main/public/app/plugins/panel/canvas/components/connections/Connections.tsx#L527

and

When modifying an existing vertex...
https://github.com/grafana/grafana/blob/main/public/app/plugins/panel/canvas/components/connections/Connections.tsx#L406-L407

One solution would be to set the coordinate to zero instead of infinity to avoid the render error; however, this would not allow the connector shape to be modified in any meaningful way. Instead, I think invoking a minimum width and height of the relative vertex coordinate system would be a better approach. This would also reduce unexpected behavior when elements are being moved around.

### PR DESCRIPTION
**Pull request by drew08t**
Fix issue where division by zero was occurring during flowchart connection edit for cases where source and target anchors have the same x or y coordinate. This also goes after unexpected behavior when dragging an element.

- [x] Remove division by zero
- [x] ~~Add minimum constraint for x and y delta~~
- [x] New vertices not lined up properly
- [x] Apply fix to existing vertices
- [x] Temporary edit lines need to be adjusted
- [x] Explore fixing for all cases (even those beyond minimum x and y delta region) #85295
- [x] Update vertex handling to use new persistant original coordindate system
- [x] Fix curves
- [ ]~~Ensure tangency for first segment when using curves~~ https://github.com/grafana/grafana/issues/86210
- [x] ~~Code clean-up~~ https://github.com/grafana/grafana/issues/85013

Fixes #85031, #85295

### CODE REVIEW

**Review by adela-almasan — APPROVED** [review]
I think it looks good! 🎉 we're already discussed about fixing existing connections in a follow-up, so this should be fine!

**Review by baldm0mma — APPROVED** [review]
LGTM. Your arithmetic operational skills are unparalleled!!!

File: public/app/plugins/panel/canvas/components/connections/ConnectionSVG.tsx [hunk 1]
```diff
@@ -224,12 +232,14 @@ export const ConnectionSVG = ({
 
               if (index === 0) {
                 // For first vertex
-                addVertices.push(calculateMidpoint(0, 0, x, y));
+                addVertices.push(
+                  calculateMidpoint((x1 - xStart) / (xEnd - xStart), (y1 - yStart) / (yEnd - yStart), x, y)
```
Comment (nmarrs) [inline public/app/plugins/panel/canvas/components/connections/ConnectionSVG.tsx:236]: might be a little cleaner to break out these operations and assign them to a variable to be passed into this helper function

both here and [line 331 of this file](https://github.com/grafana/grafana/pull/85443/files#diff-88e95aa7a0e12a09f7ec48f2ad2f5289e30d62f438e5948897c3b18e1d86ca1dR331)

**Review by nmarrs — APPROVED** [review]
Overall works well / better than before 😁 

I did notice a bit of strange behavior with vertexes moving when perhaps it doesn't make sense (note the vertexes moving back and forward connected to the "Jobs" and "Alerts" boxes) but in its current form it is better than before so 🤷‍♂️ The only thing I struggle a bit with is adding more to the connection save model (i.e. I wonder if there's another way to access this information / calculate it as needed vs codify it for every connection)


https://github.com/grafana/grafana/assets/22381771/df6b1078-b9e0-4201-b8b3-21d83182316b

File: public/app/plugins/panel/canvas/components/connections/ConnectionSVG.tsx (hunk 1 above)
Comment (drew08t) [reply public/app/plugins/panel/canvas/components/connections/ConnectionSVG.tsx:236]: Good suggestion! I'll be sure to incorporate this during clean up work: https://github.com/grafana/grafana/issues/85013

**Comment by drew08t** [comment]
> I did notice a bit of strange behavior with vertexes moving

Yeah, this is a great example of old connections not storing the original source / target coordinates. We should be able to swap missing original coordinates with current coordinates to prevent this: https://github.com/grafana/grafana/issues/86379

**Comment by drew08t** [comment]
> The only thing I struggle a bit with is adding more to the connection save model (i.e. I wonder if there's another way to access this information / calculate it as needed vs codify it for every connection)

Yeah, I have similar feelings on this; however, I can't see a way to efficiently handle this unless we adopt an absolute top/left coordinate system. Currently the coordinate vertices are defined relative to the source / target coordinates. To remove this requirement, they would need to be defined absolutely; however, I anticipate this being tricky when dealing with alternate layout options. Happy to explore this further as a storage optimization follow up!

**Pull request merged**